lib/chess/pieces/piece.rb

```diff
@@ -0,0 +1,116 @@
+require_relative '../chess'
+require_relative '../step'
+require 'colorize'
+require 'colorized_string'
+require 'byebug'
+
+module Chess
+  class Piece
+    attr_reader :symbol, :team, :current_step
+
+    Vector = Struct.new(:enabled, :deltas)
+
+    def initialize(position_algebraic, symbol, team, captured = false)
+      @current_step = Step.new(position_algebraic)
+      @symbol = symbol
+      @captured = captured
+      @team = team
+      @enemies_team = ([WHITE_TEAM, BLACK_TEAM] - [@team]).first
```

Comment:
These two attributes depend one on the other, leading to duplication. It would be better to have `enemies_team` as a method instead of an attribute, so that it computes itself depending on the value `@team`